### PR TITLE
layers: Check StoreOpDontCare then LoadOpLoad

### LINKED ISSUE #293: Warn if attachment STORE_OP_DONT_CARE is followed by LOAD_OP_LOAD
Issue was found in third party game. It could be traced with validation layer.

Game uses same attachment in first render pass with STORE_OP_DONT_CARE and later in another pass it tries to use it with LOAD_OP_LOAD.

On some gfx vendors it works, on some corruptions are visible.

Edit:
Validation layers could print some warning :)

### PR DESCRIPTION
**Pull request by paul-lunarg**
Add warning message when loading an image in a renderpass with LOAD_OP_LOAD when the last operation on the image was a discard with STORE_OP_DONT_CARE, which results in undefined behaviour.
Closes #293

### CODE REVIEW

**Review by jeremyg-lunarg — COMMENTED** [review]
Other than the commented out code this LGTM

File: tests/vklayertests_best_practices.cpp [hunk 1]
```diff
@@ -1771,4 +1771,124 @@ TEST_F(VkBestPracticesLayerTest, RenderPassClearValueCountHigherThanAttachmentCo
...
+
+    // Setup finished
+    m_errorMonitor->VerifyNotFound();
+
+    // TEST :
+    m_errorMonitor->SetDesiredFailureMsg(kWarningBit, kVUID_BestPractices_StoreOpDontCareThenLoadOpLoad);
+
+    // This should give a warning
+    vk::QueueSubmit(m_device->m_queue, 1, &submit_info, VK_NULL_HANDLE);
+
```
Comment (Tony-LunarG) [inline tests/vklayertests_best_practices.cpp:1892]: I think you need to either wait on a fence given to QueueSubmit, or do a QueueWaitIdle here or all the object destructors will happen on objects still considered in flight in the command buffer.

Comment (paul-lunarg) [reply tests/vklayertests_best_practices.cpp:1892]: Wasn't sure if it was necessary, but it is better with a QueueWaitIdle

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Review by jeremyg-lunarg — APPROVED** [review]

**Review by ncesario-lunarg — APPROVED** [review]
LGTM, nice work!

**Pull request merged**